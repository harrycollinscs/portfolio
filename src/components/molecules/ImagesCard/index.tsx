import React from 'react'
import { Slide } from 'react-slideshow-image'
import Card from '../../atoms/Card'
import styled from 'styled-components'

const Image = styled.img`
  height: 100%;
  width: 100%;
  border-radius: 50%;
  pointer-events: none;
`

const ImagesCard = ({ images }: any) => (
  <Card noPadding circle bgColor='transparent'>
    <Slide prevArrow={<></>} nextArrow={<></>} transitionDuration={500}>
      {images?.map((image: any, index: number)=> (
        <div className="each-slide" key={index}>
          <Image src={image} alt="sliding"/>
        </div>
      ))} 
    </Slide>
  </Card>
)

export default ImagesCard