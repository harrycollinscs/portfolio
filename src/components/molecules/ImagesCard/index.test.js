import React from 'react'
import renderer from 'react-test-renderer'
import ImagesCard from './index'

it('renders correctly', () => {
  const tree = renderer.create(<ImagesCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
