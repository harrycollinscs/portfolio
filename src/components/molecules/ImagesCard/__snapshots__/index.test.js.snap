// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bcXHqe kEQVfj"
>
  <div
    aria-roledescription="carousel"
    dir="ltr"
  >
    <div
      className="react-slideshow-container"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="react-slideshow-wrapper slide "
      >
        <div
          className="images-wrap"
          style={
            Object {
              "transform": "translate(-0px)",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
